Common/src/main/java/me/egg82/tfaplus/services/InternalAPI.java

```diff
@@ -95,36 +96,25 @@ public String registerHOTP(UUID uuid, long codeLength, long initialCounter, Jedi
 
         // SQL
         HOTPData result = null;
-        try {
-            if (sqlType == SQLType.MySQL) {
-                result = MySQL.updateHOTP(sql, storageConfigNode, uuid, codeLength, initialCounter, key).get();
-            } else if (sqlType == SQLType.SQLite) {
-                result = SQLite.updateHOTP(sql, storageConfigNode, uuid, codeLength, initialCounter, key).get();
-            }
-        } catch (ExecutionException ex) {
-            logger.error(ex.getMessage(), ex);
-        } catch (InterruptedException ex) {
-            logger.error(ex.getMessage(), ex);
-            Thread.currentThread().interrupt();
-        }
+        result = getHotpData(uuid, codeLength, initialCounter, key, result);
 
         if (result == null) {
             return null;
         }
 
         // Redis
-        Redis.update(result, redisPool, redisConfigNode);
+        Redis.update(result, ConfigUtil.getRedisPool(), ConfigUtil.getRedisConfigNode());
 
         // RabbitMQ
-        RabbitMQ.broadcast(result, rabbitConnection);
+        broadcastResult(result);
```

Comment:
Redis & Rabbit broadcast/update should probably be in one method, if we're going to make it a method anyway.